weave

```diff
@@ -1093,6 +1093,61 @@ DOCKER_VERSION_PATCH=$(echo "$DOCKER_VERSION" | cut -d. -f 3)
 COMMAND=$1
 shift 1
 
+deprecation_warning() {
+    echo "Warning: $1 is deprecated; please use $2" >&2
+}
+
+deprecation_warnings() {
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            -password|-password=*)
+                deprecation_warning "-password" "--password"
+                [ "$1" == "-password" ] && shift
+                ;;
+            --password)
+                shift
+                ;;
+            -nickname|-nickname=*)
+                deprecation_warning "-nickname" "--nickname"
+                [ "$1" == "-nickname" ] && shift
+                ;;
+            --nickname)
+                shift
+                ;;
+            -nodiscovery|--nodiscovery)
+                deprecation_warning "[-]-nodiscovery" "--no-discovery"
+                ;;
+            -iprange|--iprange|-iprange=*|--iprange=*)
+                deprecation_warning "[-]-iprange" "--ipalloc-range"
+                [ ${1#--} == "iprange" ] && shift
+                ;;
+            --ipalloc-range)
+                shift
+                ;;
+            -ipsubnet|--ipsubnet|-ipsubnet=*|--ipsubnet=*)
+                deprecation_warning "[-]-ipsubnet" "--ipalloc-default-subnet"
+                [ ${1#--} == "ipsubnet" ] && shift
+                ;;
+            --ipalloc-default-subnet)
+                shift
+                ;;
+            -initpeercount|--initpeercount|-initpeercount=*|--initpeercount=*)
+                deprecation_warning "[-]-initpeercount" "--init-peer-count"
+                [ ${1#--} == "initpeercount" ] && shift
+                ;;
+            --init-peer-count)
+                shift
+                ;;
+            -no-default-ipam|--no-default-ipam)
+                deprecation_warning "[-]-no-default-ipam" "--no-default-ipalloc"
+                ;;
+        esac
```

Comment:
> Why is this intercepting --<new-option>?

As in `--ipalloc-default-subnet)` ? Yes, just to skip over the next argument. I think it's still worth attempting accidental parsing of values as options, even if it can't be infallible.